lib/Transforms/SimplifyAffineStructures.cpp

```diff
@@ -102,8 +102,14 @@ void SimplifyAffineStructures::runOnFunction() {
     }
   });
 
-  // Turn memrefs' non-identity layouts maps into ones with identity.
-  func.walk([](AllocOp op) { normalizeMemRef(op); });
+  // Turn memrefs' non-identity layouts maps into ones with identity. Collect
+  // alloc ops first and then process since normalizeMemRef replaces/erases ops
+  // during memref rewriting.
+  SmallVector<AllocOp, 4> allocOps;
+  func.walk([&](AllocOp op) { allocOps.push_back(op); });
+  for (auto allocOp : llvm::make_early_inc_range(allocOps)) {
```

Comment:
You don't need `make_early_inc_range` here as you're not modifying the `allocOps` vector itself.

Reply:
Fixed, thanks.